AutoCollection/ClientRequestParser.ts

```diff
@@ -48,15 +53,26 @@ class ClientRequestParser extends RequestParser {
         let dependencyName = this.method.toUpperCase() + " " + url.format(urlObject);
 
         let remoteDependency = new ContractsModule.Contracts.RemoteDependencyData();
-        remoteDependency.target = urlObject.hostname;
+
+        if (this.targetIKeyHash) {
+            // Currently the service supports a dependency type name of ApplicationInsights,
+            // but not a dependency kind. So the dependency kind here is still just HTTP.
+            remoteDependency.dependencyKind = ContractsModule.Contracts.DependencyKind.Http;
+            remoteDependency.dependencyTypeName = ContractsModule.Contracts.DependencyKind
+                [ContractsModule.Contracts.DependencyKind.ApplicationInsights];
```

Comment:
The space is unfortunate, because it means the allowed values can no longer be represented as a TypeScript enum. So we'll have to deprecate the whole DependencyKind enum.

And then will that be a breaking change to the [`trackDependency`](https://github.com/Microsoft/ApplicationInsights-node.js/blob/develop/Library/Client.ts#L148) API?